### PR TITLE
API-3139: add proxy support to com.openx.oauth.client.Helper

### PR DESCRIPTION
**Pull request by openx-luis**

### CODE REVIEW

File: src/main/java/com/openx/oauth/client/Helper.java [hunk 1]
```diff
@@ -70,28 +72,30 @@ public Helper(String url, String username, String password, String token) {
...
-    public String doLogin() throws UnsupportedEncodingException, IOException {
-        return doLogin(null);
+    public Helper(HttpHost proxy, String url, String username, String password, String token) {
+        this.proxy = proxy;
+        this.url = url;
+        this.username = username;
+        this.password = password;
+        this.token = token;
+        this.cookieStore = null;
     }
```
Comment (zhentao) [inline src/main/java/com/openx/oauth/client/Helper.java]: use this(url, username, password, token)

File: src/main/java/com/openx/oauth/client/Helper.java [hunk 2]
```diff
@@ -47,6 +47,7 @@
  */
 public class Helper {
 
+    protected HttpHost proxy;
     protected HttpURLConnection connection;
```
Comment (zhentao) [inline src/main/java/com/openx/oauth/client/Helper.java]: remove it since it is not being used.

Comment (openx-luis) [inline src/main/java/com/openx/oauth/client/Helper.java]: It's used in the `openxHttpClientFactory()` function, though.

File: src/main/java/com/openx/oauth/client/Helper.java (hunk 1 above)
Comment (openx-luis) [inline src/main/java/com/openx/oauth/client/Helper.java]: ok. I made a second PR to use that instead.

File: src/main/java/com/openx/oauth/client/Helper.java (hunk 2 above)
Comment (zhentao) [inline src/main/java/com/openx/oauth/client/Helper.java]: I meant this one:

protected HttpURLConnection connection;

**Pull request merged**